mlflow/projects/docker.py

```diff
@@ -114,7 +139,7 @@ def _create_docker_build_ctx(work_dir, dockerfile_contents):
             output_filename=result_path, source_dir=dst_path, archive_name=_PROJECT_TAR_ARCHIVE_NAME
         )
     finally:
-        shutil.rmtree(directory)
+        shutil.rmtree(directory, onerror=handle_readonly)
```

Comment:
could you add a unit test to verify the handle_readonly works ?

Reply:
Done!